src/Microsoft.Identity.Client/ApiEvent.cs

```diff
@@ -83,5 +109,20 @@ public bool WasSuccessful
             set => this[ConstWasSuccessful] = value.ToString().ToLower();
             get => this[ConstWasSuccessful] == true.ToString().ToLower();
         }
+
+        public string CorrelationId
```

Comment:
Yes, emptystring or null string is OK, they won't cause exception for TenantId and UserId.